### PR TITLE
TECH implement and test contact list endpoints

### PR DESCRIPTION
**Pull request by alafanechere**
Emarsys has a feature to [add contacts to lists](https://dev.emarsys.com/v2/contact-lists). 

I implemented client side communication with two API  endpoints related to this feature : 
* `POST api/v2/contactlist/` : Create a new contact list
* `POST api/v2/contactlist/{listId}/add/` : Add a contact to a contact list

More contact list endpoints are available in the API but not required for our use cases yet.

### CODE REVIEW

File: pymarsys/contact_list.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    def create(self, name, key_id=3, with_contacts_id=None, description=None):
+        """
+        Create a contact list.
+        https://dev.emarsys.com/v2/contact-lists/create-a-contact-list
+        :param name: name of the list to create
+        :param key_id: Key which identifies the contact. This can be a field
+        id, id, uid or eid. If left empty, the email address (field ID 3) will
+        be used by default.
+        :param with_contacts: list of key's value to add to the list. e.g. ['squirrel@squirrelmail.com',]
```
Comment (MohamedJeffal) [inline pymarsys/contact_list.py]:
```suggestion
        :param with_contacts_id: list of key's value to add to the list. e.g. ['squirrel@squirrelmail.com',]
```

Small nitpick, but shouldn't the last fragment of the name be plural, eg `with_contact_ids`?

File: pymarsys/contact_list.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        Examples:
+        If you want to add two squirrel1@squirrelmail.com to the list test_list:
+        >>> client.contact_list.add('test_list', ['squirrel1@squirrelmail.com',])
+        {'data': {'inserted_contacts': 123}, 'replyCode': 0, 'replyText': 'OK'}
+        """
+
+        payload = {"key_id": key_id, "external_ids": contacts_id}
+
+        endpoint = f"{self.endpoint}{list_id}/add/"
```
Comment (MohamedJeffal) [inline pymarsys/contact_list.py]: Aren't there utilities in the python stdlib to resolve urls fragments?

File: tests/test_contact_list.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            responses.POST,
+            urljoin(EMARSYS_URI, CONTACT_LIST_ENDPOINT),
+            json=EMARSYS_CONTACT_LIST_CREATE_RESPONSE,
+            status=200,
+            content_type="application/json",
+        )
+        connection = SyncConnection(TEST_USERNAME, TEST_SECRET)
+        contact_list = ContactList(connection)
+        response = contact_list.create("test_list")
+        assert response == EMARSYS_CONTACT_LIST_CREATE_RESPONSE
```
Comment (MohamedJeffal) [inline tests/test_contact_list.py:49]: What do you think of also testing some error cases?

**Review by MohamedJeffal — APPROVED** [review]
We left a few comments, nice work otherwise! LGTUs 👍

File: pymarsys/contact_list.py (hunk 2 above)
Comment (alafanechere) [reply pymarsys/contact_list.py]: I could make use of urlparse's `urljoin` but it joins only a base url and a fragment, it cannot handle multiple fragment.

File: pymarsys/contact_list.py (hunk 1 above)
Comment (alafanechere) [reply pymarsys/contact_list.py]: 👍

File: tests/test_contact_list.py (hunk 3 above)
Comment (alafanechere) [reply tests/test_contact_list.py:49]: I think it's a good idea, but the rest of the package does not handle API responses errors. As far as I am not aware of the global client usage of this package I'm not sure of the kind of error handling that is expected and guess it is handled client side.

**Pull request merged**